packages/samples/headless-ssr/src/app/generic/hooks/search-parameters.ts

```diff
@@ -0,0 +1,111 @@
+'use client';
+
+import {SearchParameters} from '@coveo/headless';
+import {
+  SearchParameterManager,
+  SearchParameterManagerState,
+} from '@coveo/headless/ssr';
+import {useCallback, useEffect, useMemo, useReducer, useState} from 'react';
+import {CoveoNextJsSearchParametersSerializer} from '../common/search-parameters-serializer';
+
+interface UseSyncSearchParametersProps {
+  ssrState: SearchParameterManagerState;
```

Comment:
Would implementors have to do something similar as the following as well? Can we package some of this as utils for reuse?

Reply:
We could, but those would probably be very framework-specific, so we'd probably need to export them into yet another package (e.g., `headless-next`).